### PR TITLE
abstracted component for icon

### PR DESCRIPTION
**Pull request by firwer**
Closes Icons #6

Added abstracted component for icons that will take in the MUI icon and style it based on the specs.

### CODE REVIEW

File: assets/icons.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  Icon: any
+}
+// Optional props
+interface IconOptionalProps {
+  width: number
+  height: number
+  color: string
+}
+// Combine required and optional props to build the full prop interface
+interface IconProps extends IconRequiredProps, IconOptionalProps {}
```
Comment (wilinetan) [inline assets/icons.tsx]: this one can just be `type IconProps = IconRequiredProps & IconOptionalProp`

File: assets/icons.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  height: 50,
+  color: 'black',
+}
+// Use the full props within the actual component
+export const MRIcon = (props: IconProps) => {
+  const { Icon, width, height, color } = props
+  return <Icon style={{ width: width, height: height, color: color }} />
+}
+// Be sure to set the default props
+MRIcon.defaultProps = defaultProps
```
Comment (wilinetan) [inline assets/icons.tsx]: what does this do? does this pass a `defaultProps` prop to `MRIcon`?

File: assets/icons.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// Required props
+interface IconRequiredProps {
+  Icon: any
+}
+// Optional props
+interface IconOptionalProps {
+  width: number
+  height: number
+  color: string
+}
```
Comment (wilinetan) [inline assets/icons.tsx]: for optional props
```
interface IconOptionalProps {
  width?: number
  height?: number
  color?: string
}
```

File: assets/icons.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// Required props
```
Comment (wilinetan) [inline assets/icons.tsx]: move this to `components/MRIcon/index.tsx`. assets folder is usually for images

File: assets/icons.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+// Combine required and optional props to build the full prop interface
+interface IconProps extends IconRequiredProps, IconOptionalProps {}
+// Use the optional prop interface to define the default props
+const defaultProps: IconOptionalProps = {
+  width: 50,
+  height: 50,
+  color: 'black',
+}
+// Use the full props within the actual component
+export const MRIcon = (props: IconProps) => {
```
Comment (wilinetan) [inline assets/icons.tsx]: I would prefer 
```
const MRIcon = (props: IconProps) => {...}

export default MRIcon
```

**Review by wilinetan — COMMENTED** [review]
can help to check if u pass it any of the mui icon, will it work as expected?

File: assets/icons.tsx (hunk 2 above)
Comment (firwer) [reply assets/icons.tsx]: Yes correct

**Comment by firwer** [comment]
> 
Used internal main page as an example

Default Settings: Black, w50, h50
image
image

Green, w100, h100
image
image

**Comment by wilinetan** [comment]
move the file to `components/MRIcon/index.tsx` since this is a component. other than that looks good, thanks! can merge after changing file location

**Review by wilinetan — APPROVED** [review]

**Pull request merged**